### PR TITLE
Lexemes database collection

### LINKED ISSUE #348: add Lexemes table to database
Add a Lexemes table to the database.

### PR DESCRIPTION
**Pull request by dwhieb**
**Related Issue:**

closes #348

**Description of Changes**

Adds a Lexemes "table" (object store) to the database.

### CODE REVIEW

File: src/services/Database.js [hunk 1]
```diff
@@ -88,8 +99,13 @@ export default class Database {
 
       store.createIndex(`abbreviation`, `abbreviation`, { unique: false });
       store.createIndex(`dateModified`, `dateModified`, { unique: false });
-      store.createIndex(`language`, `language.cid`, { unique: false });
+      store.createIndex(`language`, `language`, { unique: false });
```
Comment (dwhieb) [inline src/services/Database.js:102]: This now expects the `Text.prototype.language` property to be a String. The spec currently requires it to be an Object (a DatabaseReference), but I've opened an issue in the spec to allow for DatabaseReferences to be Objects or Strings.

File: src/services/DatabaseCollection.js [hunk 2]
```diff
@@ -31,16 +31,16 @@ class DatabaseCollection {
...
-        value:        database,
+        value:        Model,
         writable:     false,
       },
-      Model: {
+      idb: {
         configurable: true,
         enumerable:   true,
-        value:        Model,
+        value:        database,
```
Comment (dwhieb) [inline src/services/DatabaseCollection.js:43]: Reordered keys to satisfy ESLint.

File: src/services/DatabaseCollection.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ class DatabaseCollection {
       const isArrayInput = Array.isArray(data);
 
       const items = (isArrayInput ? data : [data])
-      .map(item => item instanceof this.Model ? item : new this.Model(item));
+      .map(item => (item instanceof this.Model ? item : new this.Model(item)));
```
Comment (dwhieb) [inline src/services/DatabaseCollection.js:69]: ESLint autoformatting.

**Review by vadekh — APPROVED** [review]
Looks, good. Should we add tests for Lexemes in the Database test file? Or are we not at that stage yet here?

**Comment by dwhieb** [comment]
@vadekh I don't think we need to. The logic for database collections is abstracted away from any particular collection, so testing the functionality of one should work for all of them. If anything does happen to break though we should add a test for the issue when we fix it.

**Pull request merged**